src/core/src/UsdNotificationHandler.cpp

```diff
@@ -332,6 +339,11 @@ void processCesiumRasterOverlayChanged(
     if (updateRasterOverlayAlpha) {
         updateRasterOverlayBindingsAlpha(context, rasterOverlayPath);
     }
+
+    if (updateRasterOverlayOptions) {
+        auto pNativeOverlay = pRasterOverlay->getRasterOverlay();
+        pRasterOverlay->updateRasterOverlayOptions(pNativeOverlay->getOptions());
```

Comment:
Even though `getRasterOverlay` is a public function it doesn't feel right that `UsdNotificationHandler` is accessing the native object. What about restructuring `OmniRasterOverlay` so the functions look like this:

```cpp
CesiumRasterOverlays::RasterOverlayOptions OmniRasterOverlay::createRasterOverlayOptions() const {
    CesiumRasterOverlays::RasterOverlayOptions options;
    options.ktx2TranscodeTargets = GltfUtil::getKtx2TranscodeTargets();
    setRasterOverlayOptionsFromUsd(options);
    return options;
}

void OmniRasterOverlay::updateRasterOverlayOptions() {
    const auto pRasterOverlay = getRasterOverlay();
    if (pRasterOverlay) {
        setRasterOverlayOptionsFromUsd(pRasterOverlay->getOptions());
    }
}

void OmniRasterOverlay::setRasterOverlayOptionsFromUsd(CesiumRasterOverlays::RasterOverlayOptions& options) const {
    options.showCreditsOnScreen = getShowCreditsOnScreen();
    options.maximumScreenSpaceError = getMaximumScreenSpaceError();
    options.maximumTextureSize = getMaximumTextureSize();
    options.maximumSimultaneousTileLoads = getMaximumSimultaneousTileLoads();
    options.subTileCacheBytes = getSubTileCacheBytes();
}
```

Previously:

```cpp
CesiumRasterOverlays::RasterOverlayOptions OmniRasterOverlay::getUsdOptions() {
    CesiumRasterOverlays::RasterOverlayOptions options;
    setOptionsFromUsd(options);
    return options;
}

void OmniRasterOverlay::updateRasterOverlayOptions(CesiumRasterOverlays::RasterOverlayOptions& options) {
    setOptionsFromUsd(options);
}

void OmniRasterOverlay::setOptionsFromUsd(CesiumRasterOverlays::RasterOverlayOptions& options) {
    options.showCreditsOnScreen = getShowCreditsOnScreen();
    options.ktx2TranscodeTargets = GltfUtil::getKtx2TranscodeTargets();
    options.maximumScreenSpaceError = getMaximumScreenSpaceError();
    options.maximumTextureSize = getMaximumTextureSize();
    options.maximumSimultaneousTileLoads = getMaximumSimultaneousTileLoads();
    options.subTileCacheBytes = getSubTileCacheBytes();
}
```

Reply:
Yes, I like that better. 

I think `updateRasterOverlayOptions` should also be `const`.

I wonder if it makes sense to move `OmniRasterOverlay::getRasterOverlay()` from `public` to `protected`? The only line that calls it is `OmniTileset` in the code below. We could `friend` the `add`. It seems nice to enforce Omni* classes only talking to other Omni* classes. Then the only way to get to a Native object is via its Omni class.

```
    for (const auto& boundRasterOverlayPath : boundRasterOverlayPaths) {
        const auto pOmniRasterOverlay = _pContext->getAssetRegistry().getRasterOverlay(boundRasterOverlayPath);
        if (pOmniRasterOverlay) {
            pOmniRasterOverlay->reload();
            const auto pNativeRasterOverlay = pOmniRasterOverlay->getRasterOverlay();
            if (pNativeRasterOverlay) {
                _pTileset->getOverlays().add(pNativeRasterOverlay);
            }
        }
    }
```

Reply:
Sure, a friend class works. I don't tend to use them but I don't have an issue with it.

Reply:
I added two friend methods instead of a class so `OmniTileset` wouldn't have access to all the `OmniRasterOverlay` internals. The friend class would be a clean one-liner but less locked down.